src/util/bip32.rs

```diff
@@ -46,11 +46,73 @@ impl Default for Fingerprint {
     fn default() -> Fingerprint { Fingerprint([0; 4]) }
 }
 
+/// Script type for addresses
+#[derive(PartialEq, Eq, Clone, Copy, Debug)]
+pub enum ScriptType {
+    /// Pay to pubkey hash
+    P2pkh,
+    /// Pay to witness pubkey hash
+    P2wpkh,
+    /// Pay to witness pubkey hash, wrapped in pay to script hash
+    P2shP2wpkh,
```

Comment:
Also, quite confusing names. May be `Legacy`, `SegWitV0`, and `SegWitLegacy` will work better?